src/main/java/core/basesyntax/model/FruitTransaction.java

```
@@ -0,0 +1,29 @@
+package core.basesyntax.model;
+
+public class FruitTransaction {
+    private Operation operation;
+    private String fruit;
+    private int quantity;
+
+    private FruitTransaction(Operation operation, String fruit, int quantity) {
+        this.operation = operation;
+        this.fruit = fruit;
+        this.quantity = quantity;
+    }
+
+    public static FruitTransaction create(Operation operation, String fruit, int quantity) {
+        return new FruitTransaction(operation, fruit, quantity);
+    }
```

Add validation to the factory/constructor: disallow null `operation`, null/empty `fruit`, and negative `quantity`. The project checklist requires that quantities be non-negative and invalid data should cause a RuntimeException (quote: "Quantity must be a non-negative integer; invalid quantity should cause a RuntimeException."). Fail fast with informative messages so invalid transactions are never created.